bot/cogs/bot.py

```diff
@@ -73,38 +74,60 @@ def __init__(self, bot: AutoShardedBot):
+            if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
                     try:
-                        # Some users formatted multi-line code using `'sip("
-                        # This fixes that by treating them as nothing.
-                        if "`" in msg.content:
-                            msg.content = msg.content[:-1][1:]
-                        tree = ast.parse(msg.content)
-
                         # Attempts to parse the message into an AST node.
                         # Invalid Python code will raise a SyntaxError.
-                        if not all(isinstance(node, ast.Expr) for node in tree.body):
+                        content = self.codeblock_stripping(msg.content)
+                        if not content:
+                            return
 
-                            # Multiple lines of single words could be interpreted as expressions.
-                            # This check is to avoid all nodes being parsed as expressions.
-                            # (e.g. words over multiple lines)
-                            howto = ("Please use syntax highlighted blocks, as it makes "
-                                     "your code more legible for other users.\n"
-                                     "\nTo do this, you should input your content like this:\n"
-                                     "\n\`\`\`python\n"
-                                     "print(\"Hello world!\")\n"
-                                     "\`\`\`\n"
-                                     "\nThis will result in the following:\n"
-                                     "```python\n"
-                                     "print(\"Hello world!\")"
-                                     "```"
-                            )  # noqa. E124
-                            information = Embed(title="Code formatting", description=howto)
-                            await msg.channel.send(embed=information)
-                            self.previous_format_times[msg.channel.id] = time.time()
+                        tree = ast.parse(content)
+
+                        # Multiple lines of single words could be interpreted as expressions.
+                        # This check is to avoid all nodes being parsed as expressions.
+                        # (e.g. words over multiple lines)
+                        if not all(isinstance(node, ast.Expr) for node in tree.body):
+                            codeblock_tag = await self.bot.get_cog("Tags").get_tag_data("codeblock")
+                            howto = (f"Hey {msg.author.mention}!\n I noticed you were trying to paste code into this "
+                                     "channel. Discord supports something called Markdown, which allows you to make "
+                                     "beautiful code blocks with Python syntax highlighting!\n"
+                                     f"{codeblock_tag['tag_content']}")
```

Comment:
Actually, when you do these implicit joins you don't need f's on more than the very first string. The others will get implicit f's when they join the first one. so this f does nothing.

Reply:
nvm I'm stupid.